### PR TITLE
doc: update Hv CPU virt HLD

### PR DESCRIPTION
**Pull request by tw4452852**
This PR contains the following updates in HLD:

- CPUID virtualization.
- CR virtualization.
- IO/MMIO virtualization.
- MSR virtualization.

### CODE REVIEW

**Comment by acrnsi** [comment]
Can one of the admins verify this patch?

**Comment by leilu** [comment]
OK to verify

File: doc/developer-guides/hld/hv-cpu-virt.rst [hunk 1]
```diff
@@ -906,9 +910,13 @@ pass-through directly:
      - "
 
    * - MSR_IA32_TIME_STAMP_COUNTER
-     - TIme-stamp counter
+     - Time-stamp counter
      - work with VMX_TSC_OFFSET_FULL to emulate virtual TSC
 
+   * - MSR_IA32_APIC_BASE
+     - APIC base address
+     - emulateds with vlapic
```
Comment (lyan3) [inline doc/developer-guides/hld/hv-cpu-virt.rst]: typo: emulateds -> emulates

File: doc/developer-guides/hld/hv-cpu-virt.rst [hunk 2]
```diff
@@ -895,6 +895,10 @@ pass-through directly:
      - TSC target of local APIC's TSC deadline mode
      - emulates with vlapic
 
+   * - MSR_IA32_TSC_ADJUST
+     - TSC adjustment of local APIC's TSC deadline mode
+     - emulates with vlapic
+
```
Comment (lyan3) [inline doc/developer-guides/hld/hv-cpu-virt.rst]: It's better to move the MSR_IA32_TSC_ADJUST ahead to keep the same order of MSRs as the source code.

File: doc/developer-guides/hld/hv-cpu-virt.rst [hunk 3]
```diff
@@ -942,6 +970,18 @@ pass-through directly:
      - "
      - "
 
+   * - MSR_IA32_X2APIC_*
+     - x2APIC related MSRs
```
Comment (lyan3) [inline doc/developer-guides/hld/hv-cpu-virt.rst]: it's better to add the offset range of X2apic related MSRs.

**Comment by acrnsi** [comment]
OK to verify

**Pull request merged**